src/v/storage/segment_deduplication_utils.cc

```diff
@@ -184,56 +184,61 @@ ss::future<index_state> deduplicate_segment(
     }
     auto rdr = internal::create_segment_full_reader(
       seg, cfg, probe, std::move(read_holder));
+
+    auto segment_last_offset = seg->offsets().get_committed_offset();
```

Comment:
In general with this PR, could you please move the details about context from the cover letter into the corresponding commit messages? The commit message here reads like this isn't a functional change, while the CL mentions there is a change here.